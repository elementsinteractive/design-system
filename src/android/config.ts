import { AndroidConfig } from '../design.types'
import { TEMPLATES } from './constants'
import { AndroidJsonOptions } from './types'

export const createConfig = (input: string, options: AndroidJsonOptions) => {
  const { theme } = options

  const { colorsTemplate, spacingsTemplate, typographyTemplate } =
    TEMPLATES(theme)

  return {
    source: [input],
    platforms: {
      android: {
        transformGroup: 'tokens-android',
        files: [typographyTemplate, spacingsTemplate, colorsTemplate].map(
          template => ({
            format: template.formatter,
            destination: template.destination,
            filter: template.filter,
          }),
        ),
      },
    },
  }
}
