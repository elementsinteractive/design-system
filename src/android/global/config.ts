import { AndroidGlobalOptions } from '../types'
import { GLOBAL_TEMPLATES } from './constants'

export const createGlobalConfig = (
  input: string,
  options: AndroidGlobalOptions,
) => {
  const { spacingsTypeTemplate, colorsTypeTemplate } = GLOBAL_TEMPLATES(options)

  return {
    source: [input],
    platforms: {
      android: {
        transformGroup: 'tokens-android',
        files: [spacingsTypeTemplate, colorsTypeTemplate].map(template => ({
          format: template.formatter,
          destination: template.destination,
          filter: template.filter,
        })),
      },
    },
  }
}
