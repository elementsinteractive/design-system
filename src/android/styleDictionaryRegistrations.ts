import StyleDictionaryPackage from 'style-dictionary'
import { AndroidConfig } from '../design.types'
import { useTemplate } from '../utils'
import { TEMPLATES } from './constants'
import { colorFormatter } from './formatters/colorFormatter'
import { spacingFormatter } from './formatters/spacingFormatter'
import { typographyFormatter } from './formatters/typographyFormatter'

export const styleDictionaryRegistrations = (config: AndroidConfig) => {
  const { packageName, destination } = config

  const templateInfo = TEMPLATES(destination)

  const {
    headerTemplate,
    colorsTemplate,
    spacingsTemplate,
    typographyTemplate,
  } = templateInfo

  const header = useTemplate(headerTemplate)()

  /**
   * Formatters
   */
  StyleDictionaryPackage.registerFormat({
    name: typographyTemplate.formatter,
    formatter: typographyFormatter({
      template: typographyTemplate.source,
      header,
      packageName,
    }),
  })

  StyleDictionaryPackage.registerFormat({
    name: spacingsTemplate.formatter,
    formatter: spacingFormatter({
      template: spacingsTemplate.source,
      header,
      packageName,
    }),
  })

  StyleDictionaryPackage.registerFormat({
    name: colorsTemplate.formatter,
    formatter: colorFormatter({
      template: colorsTemplate.source,
      header,
      packageName,
    }),
  })

  /**
   * Filters
   */
  const registerFilter = (name, type) => {
    StyleDictionaryPackage.registerFilter({
      name,
      matcher: token => token.type === type,
    })
  }

  registerFilter(typographyTemplate.filter, 'typography')
  registerFilter(spacingsTemplate.filter, 'spacing')
  registerFilter(colorsTemplate.filter, 'color')

  StyleDictionaryPackage.registerTransformGroup({
    name: 'tokens-android',
    transforms: ['name/ti/camel'],
  })
}
