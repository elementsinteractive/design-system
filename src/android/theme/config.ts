import { THEME_TEMPLATES } from './constants'
import { AndroidThemeOptions } from '../types'

export const createThemeConfig = (
  input: string,
  options: AndroidThemeOptions,
) => {
  const {
    paletteTemplate,
    spacingsTemplate,
    typographyTemplate,
    colorTokensTemplate,
  } = THEME_TEMPLATES(options)

  return {
    source: [input],
    platforms: {
      android: {
        transformGroup: 'tokens-android',
        files: [
          typographyTemplate,
          spacingsTemplate,
          paletteTemplate,
          colorTokensTemplate,
        ].map(template => ({
          format: template.formatter,
          destination: template.destination,
          filter: template.filter,
        })),
      },
    },
  }
}
