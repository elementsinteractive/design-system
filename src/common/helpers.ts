import fs from 'fs-extra'
import { readFileSync } from 'fs'

export const clearOutput = path => fs.emptyDirSync(path)

export const fileExists = filePath => {
  try {
    return fs.statSync(filePath).isFile()
  } catch (err) {
    return false
  }
}

export const fileToJSON = path => fs.readJsonSync(path)

export const fileToString = path => readFileSync(path, 'utf8').toString()
