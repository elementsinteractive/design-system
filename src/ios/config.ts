import { IOSConfig } from '../design.types'
import { TEMPLATES } from './constants'
import { statSync } from 'fs'

export const createConfig = (config: IOSConfig) => {
  const { input, destination } = config

  const templateInfo = TEMPLATES(destination)

  const { colorsTemplate, spacingsTemplate, typographyTemplate } = templateInfo

  return {
    source: [input],
    platforms: {
      ios: {
        transformGroup: 'tokens-ios',
        files: [typographyTemplate, spacingsTemplate, colorsTemplate].map(
          template => ({
            format: template.formatter,
            destination: template.destination,
            filter: template.filter,
          }),
        ),
      },
    },
  }
}
