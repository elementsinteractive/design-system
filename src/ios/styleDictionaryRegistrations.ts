import StyleDictionaryPackage from 'style-dictionary'
import { IOSConfig } from '../design.types'
import { useTemplate } from '../utils'
import { TEMPLATES } from './constants'
import { colorFormatter } from './formatters/colorFormatter'
import { spacingFormatter } from './formatters/spacingFormatter'
import { typographyFormatter } from './formatters/typographyFormatter'

export const styleDictionaryRegistrations = (config: IOSConfig) => {
  const { input, themeName, destination } = config

  const templateInfo = TEMPLATES(destination)

  const {
    headerTemplate,
    colorsTemplate,
    spacingsTemplate,
    typographyTemplate,
  } = templateInfo

  const header = useTemplate(headerTemplate)()

  StyleDictionaryPackage.registerFormat({
    name: typographyTemplate.formatter,
    formatter: typographyFormatter({
      template: typographyTemplate.source,
      header,
      themeName,
    }),
  })

  StyleDictionaryPackage.registerFormat({
    name: spacingsTemplate.formatter,
    formatter: spacingFormatter({
      template: spacingsTemplate.source,
      header,
      themeName,
    }),
  })

  StyleDictionaryPackage.registerFormat({
    name: colorsTemplate.formatter,
    formatter: colorFormatter({
      template: colorsTemplate.source,
      header,
      themeName,
    }),
  })

  /**
   * Filters
   */
  const registerFilter = (name: string, type?: string) => {
    StyleDictionaryPackage.registerFilter({
      name,
      matcher: token => {
        return token.type === type
      },
    })
  }

  registerFilter(typographyTemplate.filter, 'typography')
  registerFilter(spacingsTemplate.filter, 'spacing')
  registerFilter(colorsTemplate.filter, 'color')

  StyleDictionaryPackage.registerTransformGroup({
    name: 'tokens-ios',
    transforms: ['name/ti/camel'],
  })
}
