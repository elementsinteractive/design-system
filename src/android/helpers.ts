import { readFileSync } from 'fs'
import { fileToJSON } from '../common/helpers'
import { AndroidJsonOptions } from './types'

export const readOptionsAsJson = (configPath: string) =>
  fileToJSON(configPath) as AndroidJsonOptions
