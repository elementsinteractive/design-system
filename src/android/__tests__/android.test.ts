import StyleDictionaryPackage from 'style-dictionary'
import { clearOutput, fileExists, fileToString } from '../../common/helpers'
import { createConfig } from '../config'
import { styleDictionaryRegistrations } from '../styleDictionaryRegistrations'

const androidColorFile = 'src/android/__tests__/__output/Color.kt'
const androidSpacingFile = 'src/android/__tests__/__output/Spacing.kt'
const androidTypeFile = 'src/android/__tests__/__output/Type.kt'

describe('android', () => {
  var config = {
    packageName: 'nl.makerstreet.design',
    input: 'src/android/__tests__/tokens.json',
    destination: 'src/android/__tests__/__output',
  }

  let StyleDictionaryExtended

  beforeAll(() => {
    styleDictionaryRegistrations(config)

    StyleDictionaryExtended = StyleDictionaryPackage.extend(
      createConfig(config),
    )
  })

  beforeEach(() => {
    clearOutput('src/android/__tests__/__output')
  })

  it('can setup with default config', () => {
    StyleDictionaryExtended.buildPlatform('android')

    expect(
      StyleDictionaryExtended.transformGroup['tokens-android'],
    ).toMatchSnapshot()

    const files = [androidColorFile, androidSpacingFile, androidTypeFile]

    files.forEach(path => {
      expect(fileExists(path)).toBeTruthy()
      expect(fileToString(path)).toMatchSnapshot()
    })
  })
})
