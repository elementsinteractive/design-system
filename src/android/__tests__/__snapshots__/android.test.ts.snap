// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`android can setup with default config 1`] = `
Array [
  "name/ti/camel",
]
`;

exports[`android can setup with default config 2`] = `
"package nl.makerstreet.design

/**
 * ⚠️ DO NOT MODIFY ⚠️
 * this file is auto generated using elements-design-tokens
 *
 * See https://github.com/elementsinteractive/design-tokens for more info
 */
import androidx.compose.ui.graphics.Color
import com.airbnb.android.showkase.annotation.ShowkaseColor

@ShowkaseColor(name = \\"primaryDefault\\", group = \\"material\\")
val primaryDefault = Color(0xffffffff)

@ShowkaseColor(name = \\"primaryBackground\\", group = \\"material\\")
val primaryBackground = Color(0xfff4f4f4)

@ShowkaseColor(name = \\"primaryVariant\\", group = \\"material\\")
val primaryVariant = Color(0xff000000)

@ShowkaseColor(name = \\"secondary\\", group = \\"material\\")
val secondary = Color(0xffff495b)

@ShowkaseColor(name = \\"secondaryVariant\\", group = \\"material\\")
val secondaryVariant = Color(0xfff4f4f4)

@ShowkaseColor(name = \\"surface\\", group = \\"material\\")
val surface = Color(0xffffffff)

@ShowkaseColor(name = \\"onPrimary\\", group = \\"material\\")
val onPrimary = Color(0xffffffff)

@ShowkaseColor(name = \\"onSecondary\\", group = \\"material\\")
val onSecondary = Color(0xff000000)

@ShowkaseColor(name = \\"onBackground\\", group = \\"material\\")
val onBackground = Color(0xff000000)

@ShowkaseColor(name = \\"onSurface\\", group = \\"material\\")
val onSurface = Color(0xff000000)

@ShowkaseColor(name = \\"purple\\", group = \\"material\\")
val purple = Color(0xff5048da)

@ShowkaseColor(name = \\"white\\", group = \\"material\\")
val white = Color(0xffffffff)

@ShowkaseColor(name = \\"pink\\", group = \\"material\\")
val pink = Color(0xffff495b)

@ShowkaseColor(name = \\"black\\", group = \\"material\\")
val black = Color(0xff000000)

@ShowkaseColor(name = \\"lightGray\\", group = \\"material\\")
val lightGray = Color(0xfff4f4f4)

"
`;

exports[`android can setup with default config 3`] = `
"package nl.makerstreet.design

/**
 * ⚠️ DO NOT MODIFY ⚠️
 * this file is auto generated using elements-design-tokens
 *
 * See https://github.com/elementsinteractive/design-tokens for more info
 */

object Spacings {
  const val xxxs = 4
  const val xxs = 8
  const val xs = 12
  const val s = 16
  const val m = 24
  const val l = 32
  const val xl = 48
  const val xxl = 200
}"
`;

exports[`android can setup with default config 4`] = `
"package nl.makerstreet.design

/**
 * ⚠️ DO NOT MODIFY ⚠️
 * this file is auto generated using elements-design-tokens
 *
 * See https://github.com/elementsinteractive/design-tokens for more info
 */
import androidx.compose.material.Typography
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import com.airbnb.android.showkase.annotation.ShowkaseTypography

@ShowkaseTypography(name = \\"h1\\", group = \\"material\\")
val h1 = TextStyle(
    fontWeight = FontWeight.Light,
    fontSize = 96.sp,
    letterSpacing = (-1.5).sp,
)

@ShowkaseTypography(name = \\"h2\\", group = \\"material\\")
val h2 = TextStyle(
    fontWeight = FontWeight.Light,
    fontSize = 60.sp,
    letterSpacing = (-0.5).sp,
)

@ShowkaseTypography(name = \\"h3\\", group = \\"material\\")
val h3 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 48.sp,
)

@ShowkaseTypography(name = \\"h4\\", group = \\"material\\")
val h4 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 34.sp,
    letterSpacing = (0.25).sp,
)

@ShowkaseTypography(name = \\"h5\\", group = \\"material\\")
val h5 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 24.sp,
)

@ShowkaseTypography(name = \\"h6\\", group = \\"material\\")
val h6 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 20.sp,
    letterSpacing = (0.15).sp,
)

@ShowkaseTypography(name = \\"body1\\", group = \\"material\\")
val body1 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 16.sp,
    letterSpacing = (0.5).sp,
)

@ShowkaseTypography(name = \\"body2\\", group = \\"material\\")
val body2 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 14.sp,
    letterSpacing = (0.25).sp,
)

@ShowkaseTypography(name = \\"subtitle1\\", group = \\"material\\")
val subtitle1 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 16.sp,
    letterSpacing = (0.15).sp,
)

@ShowkaseTypography(name = \\"subtitle2\\", group = \\"material\\")
val subtitle2 = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 14.sp,
    letterSpacing = (0.1).sp,
)

@ShowkaseTypography(name = \\"button\\", group = \\"material\\")
val button = TextStyle(
    fontWeight = FontWeight.Medium,
    fontSize = 14.sp,
    letterSpacing = (1.25).sp,
)

@ShowkaseTypography(name = \\"caption\\", group = \\"material\\")
val caption = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 12.sp,
    letterSpacing = (0.4).sp,
)

@ShowkaseTypography(name = \\"overline\\", group = \\"material\\")
val overline = TextStyle(
    fontWeight = FontWeight.Normal,
    fontSize = 10.sp,
    letterSpacing = (1.5).sp,
)

val AppTypography = Typography(
    h1 = h1,
    h2 = h2,
    h3 = h3,
    h4 = h4,
    h5 = h5,
    h6 = h6,
    body1 = body1,
    body2 = body2,
    subtitle1 = subtitle1,
    subtitle2 = subtitle2,
    button = button,
    caption = caption,
    overline = overline,
)"
`;
